jsonargparse/cli.py

```diff
@@ -124,12 +126,16 @@ def _add_component_to_parser(component, parser, as_positional, fail_untyped, con
             if not added_subargs:
                 remove_actions(subparser, (ActionConfigFile, _ActionPrintConfig))
             subcommands.add_subcommand(key, subparser, help=get_help_str(getattr(component, key), parser.logger))
+    else:
+        added_args = parser.add_function_arguments(component, as_group=False, **kwargs)
```

Comment:
you could also use `callable()` to verify that this is function-like

Reply:
I added a check before so at this point it will never be anything except a callable. No point in checking it again. Still changed the code because the check was an assert which wouldn't be executed if the code is optimized.